### PR TITLE
Relation equivalence fix

### PR DESCRIPTION
**Pull request by kasper-piskorski**

### CODE REVIEW

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/Tests.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    @ClassRule
+    public static final GraphContext testSet10 = GraphContext.preLoad("testSet10.gql");
+
+    @Before
+    public void onStartup() throws Exception {
+        assumeTrue(usingTinker());
+    }
+
+    @Test
+    public void testSet10() {
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/Tests.java]: Please either rename this test or java doc what this test is actually testing. If anyone makes a change and this test breaks it will be very difficult to know what to look into. 

I know it is testing a specific graql file but what is the functionality being tested for example?

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/Tests.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        QueryBuilder iqb = testSet10.graph().graql().infer(true);
+        String queryString = "match (role1: $x, role2: $y) isa relation2;";
+        QueryAnswers answers = queryAnswers(iqb.parse(queryString));
+        Assert.assertTrue(!answers.isEmpty());
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/Tests.java]: `assertTrue` is very uninformative when tests fail. Rather check equality or some other condition but if you must use assert true please attached a string to it detailing the failed assertion. So instead of `assertTrue(booleanCondition)` do `assertTrue("Assertion failure due to x not being present or blah blah", booleanCondition)`

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/Tests.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+
+import static ai.grakn.test.GraknTestEnv.usingTinker;
+import static org.junit.Assume.assumeTrue;
+
+public class Tests{
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/Tests.java]: `Tests` is a bad name for a test class I think. It provides no indication of what is being tested. What classes related to it etc. If I make a change and break this tests I won't have an idea where to start fixing it. 

Either rename or javadoc it.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/Tests.java (hunk 2 above)
Comment (aelred) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/Tests.java]: Try this:
```
Assert.assertThat(answers, not(empty()));
```
This will report a readable error stating that it expected it to be "not empty".
`not` and `empty` are imported from `Matchers`.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 4]
```diff
@@ -29,9 +29,11 @@
 
 
 import static ai.grakn.test.GraknTestEnv.usingTinker;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.not;
 import static org.junit.Assume.assumeTrue;
 
-public class Tests{
+public class ReasoningTests {
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: What is it testing in reasoning? How is it different from `ReasonerTest`? If you can't come up with a good name which I appreciate can be hard at least javadoc and describe the purpose of this test class.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 5]
```diff
@@ -46,7 +48,7 @@ public void testSet10() {
         QueryBuilder iqb = testSet10.graph().graql().infer(true);
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: This test is still named poorly and I am still unsure of what it is testing. It is testing graql file `testSet10.gql` but what about it exactly? 

As an outsider to reasoner if this test breaks I have no idea where to look to start fixing it.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java (hunk 4 above)
Comment (kasper-piskorski) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: It tests the full reasoning process. The tests in ReasonerTest test expressivity and edge cases mostly.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java (hunk 5 above)
Comment (kasper-piskorski) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: It is one of the tests Szymon wrote. More are going to follow. These are just numbered at the moment.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java (hunk 4 above)
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: Then please detail this in java docs otherwise there is a risk of the test classes being merged.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java (hunk 5 above)
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: Each test case should still have a functional purpose. What was the purpose of Szymon writing this test? What exactly was he checking. If it is a test for testing sake then I am not sure it should be added as a unit test. It should noted to become an acceptance test or release test.

Comment (kasper-piskorski) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: It is not a test for testing sake. His tests test for different meanders of reasoning and I find them all very useful. He found some cases I didn't think of. But ok, I can come up with a more meaningful name.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java (hunk 4 above)
Comment (kasper-piskorski) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: Will do

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java (hunk 5 above)
Comment (kasper-piskorski) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: This particular test tests the fix this PR is about. It fails without the correction.

Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: All I am asking is that we detail what "meanders of reasoning" are being tested in this case. 
You say it is testing what this PR is about then say in a javadoc or the test name "testing relation equivalence by doing x"

From the outsiders perspective I have no idea why this test should exist so when I break it I might be tempted to `@Ignore` and move on. I am simply trying to prevent that from happening by having clearly stated test cases.

If you changed this to something like:

```
/**
* Tests backward chaining in a scenario with a deep ontology.
* Tests lazy loading a large dataset
* Tests Relation equivalence etc . . . 
**/
public void testSet10(){}
```

That provides a lot more information as to where to start debugging if this test breaks.
